tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -159,16 +156,16 @@ sub run_test {
 sub prepare_host {
 
     #install required packages on host
-    zypper_call '-t in pciutils nmap';                    #to run 'lspci' and 'nmap' command
+    zypper_call '-t in pciutils nmap';    #to run 'lspci' and 'nmap' command
 
-    #check IOMMU on XEN is enabled
-    if (is_xen_host()) {
-        assert_script_run "xl dmesg | grep IOMMU | grep -i enabled";
+    #check VT-d is supported in Intel x86_64 machines
+    if (script_run("grep Intel /proc/cpuinfo") == 0) {
```

Comment:
What about AMD cpu? The SR-IOV tests should cover both CPU types' HW.

Reply:
No extra action is needed for AMD cpu. It seems iommu is default enabled on all machines, but I have not found a reliable official document for it. I'll reference the doc if I find one in the future.

Reply:
Hope this can help you:
https://www.linux-kvm.org/page/How_to_assign_devices_with_VT-d_in_KVM

Reply:
Thanks. The document seems be based on an older SLES, the dmesg output is a little different. Besides, it just describes kvm on Intel and AMD, no XEN. intel_iommu=on is needed on XEN as well. The output between KVM and Xen are different either. So here I did the minimal check(work both kvm and xen) to make sure Intel VT-d is enabled in Intel machine, no check for AMD as it seems AMD-Vi is enabled by default.